src/Solver/DantzigWolfe/DwWorker.cpp

```diff
@@ -273,11 +273,30 @@ DSP_RTN_CODE DwWorker::generateCols(
 
 			if (status == DSP_STAT_DUAL_INFEASIBLE) {
 				/** retrieve ray if unbounded */
-				std::vector<double*> rays = osi_[s]->si_->getPrimalRays(1);
+				std::vector<double*> rays;
+
+				switch(par_->getIntParam("DW/SUB/SOLVER")) {
+				case OsiCpx:
+					rays = osi_[s]->getUnbdRay();
+					break;
+				case OsiGrb:
+					// rays = osi_[s]->getUnbdRay();
+					throw CoinError("getPrimalRays not implemented in Gurobi.", "DwWorker", "DwWorker.cpp");
```

Comment:
When throwing the error, does the code print out the message that describes the solution status? If not, please add the proper message to explain the solution status.

Reply:
"Encountered unbounded subproblem. getPrimalRays not implemented in Gurobi."